opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -179,6 +245,7 @@ def __init__(
         attributes: types.Attributes = None,  # TODO
         events: typing.Sequence[Event] = None,  # TODO
         links: typing.Sequence[Link] = None,  # TODO
+        span_processor: SpanProcessor = _NO_OP_SPAN_PROCESSOR,
```

Comment:
the PR on SpanProcessor in the spec implies that the SpanProcessor would be attached to the tracer: https://github.com/open-telemetry/opentelemetry-specification/pull/205/files#r311622907

To me this feels more intuitive as well: no need to go attach a SpanProcessor to every span I create. What was the choice to attach to the Span? enable overriding of what is specified by the tracer?

Reply:
For me this looks more like an implementation choice: The span only needs to know the span processors not the whole Tracer that contains it. But since the Tracer currently switches from None/No-Op to a MultiSpanProcessor, this does indeed violate the specification for spans created before the first span processor is added.

To fix this, I suggest to always use the same MultiSpanProcessor in the Tracer right from `__init__` (I also commented something to that effect in Tracer.add_span_processor). And, of course, add a unit test against that bug 😉

Reply:
@toumorokoshi the way you mentioned is the one used in this PR.
Spans created with the `create_span` method are already attached to the span processors configured in the tracer. It is not needed to attach them later on.

@Oberon00 Although I think you are right, I also think the specification is not that clear, is it really true that SpanExporters must be called for Spans that started before they were added?

Reply:
You are right, the specification is unclear here. Then I would say, do whatever is easier to implement (but it might still make sense to add a unit test to ensure we don't change the behavior we decide for)

Reply:
> For me this looks more like an implementation choice

Correct. The Specification mentions the processor being attached to the `Tracer`, but it doesn't mention internally it *should not* be attached to `Span`s.

>  this does indeed violate the specification for spans created before the first span processor is added.

As @mauriciovasquezbernal said, I *think* the specification does not clarify this. I get the feeling the current simple approach is good enough - there's also the possible related case of `Span`s getting, if using always the current `Tracer` processor, of getting `on_end()` and not `on_start()`.

Reply:
> but it might still make sense to add a unit test to ensure we don't change the behavior we decide for

Agreed.

Reply:
In my implementation a span processor will receive `on_end` notifications from spans that were created before it was added to the tracer.  I added a test for it.

Reply:
Mmm that's an interesting thing, as we might need to mention (in the Specification) that a processor could get a `End` call without a start (or else restrict this case).

Will think about it a little bit and fill an issue about it most likely.